README.md

```diff
@@ -368,294 +396,260 @@ A device renderer instance can be configured using the `options` argument (objec
+    Enables or disables the identifiers widget. This widget can be used to set the identifiers (Android ID / IMEI) of the Android virtual device.
 
-### `fileUploadUrl`
+#### `keyboard`
 
--   **Type:** `String`
--   **Default:** `undefined`
+-   **Type:** `Boolean`
+-   **Default:** `true`
 -   **Compatibility:** `PaaS`, `SaaS`
 -   **Details:**
-    Set the file upload url, required if `fileUpload` is set to `true`.
+    Enables or disables the keyboard widget. This widget can be used to transmit keyboard keystrokes to the Android virtual device.
 
-### `clipboard`
-
-<img align="right" src="./doc/assets/ic_clipboard.svg" alt="..."></img>
+#### <img style="vertical-align: middle" src="./doc/assets/ic_keymapping.svg"> `keyboardMapping`
 
 -   **Type:** `Boolean`
 -   **Default:** `true`
+-   **Toolbar name:** `KeyboardMapping`
 -   **Compatibility:** `PaaS`, `SaaS`
 -   **Details:**
-    Enables or disables the clipboard widget. This widget can be used to forward local clipboard to the Android virtual device.
+    Enables or disables the keyboard mapping. This widget can be used to map keys with commands (e.g., tap, swipe-left, tilt, etc.).
 
-### `battery`
-
-<img align="right" src="./doc/assets/ic_battery.svg" alt="..."></img>
+#### `microphone`
 
 -   **Type:** `Boolean`
--   **Default:** `true`
--   **Compatibility:** `PaaS`, `SaaS`
+-   **Default:** `false`
+-   **Compatibility:** `PaaS`
 -   **Details:**
-    Enables or disables the battery widget. This widget can be used to set the battery level and state of the Android virtual device.
-
-### `gps`
+    Enables or disables microphone injection. This can be used to forward local microphone (or webcam audio) to the Android virtual device.
 
-<img align="right" src="./doc/assets/ic_location.svg" alt="..."></img>
+#### `mouse`
 
 -   **Type:** `Boolean`
 -   **Default:** `true`
 -   **Compatibility:** `PaaS`, `SaaS`
 -   **Details:**
-    Enables or disables the gps widget. This widget can be used to set the gps location of the Android virtual device.
-    If you want to use a visual map instead of GPS coordinates number to set the location, you must import google maps
-    library with your API key.
-
-```html
-<!-- OPTIONAL: Import google maps library with your API key to enable map positioning feature -->
-<script src="https://maps.googleapis.com/maps/api/js?key=xxxxxxxxxxxxxxxxxxxxxxxx-yyyyyyyyyyyyyy"></script>
-```
+    Enables or disables the mouse events. If you want to disable all VM interaction, please also disable `touch` and `keyboard`.
 
-### `gpsSpeedSupport`
+#### <img style="vertical-align: middle" src="./doc/assets/ic_nav_android_multiapp.svg"><img style="vertical-align: middle" src="./doc/assets/ic_nav_android_home.svg"><img style="vertical-align: middle" src="./doc/assets/ic_nav_android_back.svg"> `navbar`
```

Comment:
This image appears blank?

![image](https://github.com/user-attachments/assets/9bf50f5e-88cb-4f9c-8ad9-dd8fe49cd826)

Reply:
Ah, I understand better in light mode!

![image](https://github.com/user-attachments/assets/7d5c6943-55a1-4460-8fad-d8ada9b70ed2)

Reply:
If I understand right, these icons have been removed from the readme now?